clients/src/main/java/org/apache/kafka/clients/admin/ListConsumerGroupsOptions.java

```diff
@@ -34,13 +35,24 @@ public class ListConsumerGroupsOptions extends AbstractOptions<ListConsumerGroup
 
     private Set<ConsumerGroupState> states = Collections.emptySet();
 
+    private Set<GroupType> types = Collections.emptySet();
+
     /**
-     * If states is set, only groups in these states will be returned by listConsumerGroups()
+     * If states is set, only groups in these states will be returned by listConsumerGroups().
      * Otherwise, all groups are returned.
      * This operation is supported by brokers with version 2.6.0 or later.
      */
     public ListConsumerGroupsOptions inStates(Set<ConsumerGroupState> states) {
-        this.states = (states == null) ? Collections.emptySet() : new HashSet<>(states);
+        this.states = (states == null || states.isEmpty()) ? Collections.emptySet() : new HashSet<>(states);
+        return this;
+    }
+
+    /**
+     * If types is set, only groups of these types will be returned by listConsumerGroups().
+     * Otherwise, all groups are returned.
```

Comment:
This is interesting. Does "all groups are returned" mean that `UNKNOWN` groups are returned too?

Reply:
yes yes